File: hackerrank/problem solving/algorithms/1 - warmup/2 - simple array sum/simple_array_sum.cs
// https://www.hackerrank.com/challenges/simple-array-sum/problem?isFullScreen=true

Console.ReadLine();
Console.WriteLine(SumStdin());

// n: number of integers in the input line
// T: O(n)
// S: O(n) extra space
int SumStdin() => Console.ReadLine()!.Split().Select(int.Parse).Sum();

File: hackerrank/problem solving/algorithms/1 - warmup/2 - simple array sum/simple_array_sum.rs
// https://www.hackerrank.com/challenges/simple-array-sum/problem?isFullScreen=true

fn main() {
    let n: usize = text_io::read!();
    print!("{}", sum_stdin(n));
}

// T: O(n)
// S: O(1) extra space
fn sum_stdin(n: usize) -> i32 {
    (0..n).map(|_| -> i32 { text_io::read!() }).sum()
}
